### PR TITLE
fix: limit order custom limit price state lost

### LINKED ISSUE #8216: Custom limit price input is lost when navigating back to limit input from preview step
### Overview

Custom limit price input is lost when navigating back to limit input from preview step.

### References and additional details

https://jam.dev/c/4a8fa9d4-e615-45de-a670-2a2d6089ddc1

https://github.com/shapeshift/web/pull/8173#pullrequestreview-2466740965

### Acceptance Criteria

Custom limit and other inputs should be preserved when navigating back from an order preview.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Fixes issue where custom limit price input is lost when navigating back to limit input from preview step.

Moves `presetLimit` into redux and moves the reset of limit order config on asset change into redux, so component mount doesn't trigger a reset.

Includes an improvement to `createTradeInputBaseSlice` to allow us to compose redux actions (needed for this PR).

## Issue (if applicable)

closes #8216

## Risk

> High Risk PRs Require 2 approvals

Low risk. Review redux changes with care though.

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing

Check limit orders config is only reset when changing assets.

Worth giving spot trades a test in case the are broken for some surprise reason.

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

## Screenshots (if applicable)

Demo:
https://jam.dev/c/d1b1416d-d0d8-48f4-b6d2-c162a7801dc2

### CODE REVIEW

File: src/state/slices/tradeInputSlice/tradeInputSlice.ts [hunk 1]
```diff
@@ -24,7 +25,7 @@ const initialState: TradeInputState = {
 export const tradeInput = createTradeInputBaseSlice({
   name: 'tradeInput',
   initialState,
-  extraReducers: {
+  extraReducers: (_baseReducers: SliceCaseReducers<TradeInputState>) => ({
     // Add any reducers specific to tradeInput slice here that aren't shared with other slices
-  },
+  }),
```
Comment (woodenfurniture) [inline src/state/slices/tradeInputSlice/tradeInputSlice.ts:30]: This is why spot trades should be safe - no real change.

File: src/state/slices/limitOrderInputSlice/limitOrderInputSlice.ts [hunk 2]
```diff
@@ -52,5 +72,17 @@ export const limitOrderInput = createTradeInputBaseSlice({
...
+    },
+    setSellAsset: (state: LimitOrderInputState, action: PayloadAction<Asset>) => {
+      baseReducers.setSellAsset(state, action)
+      resetLimitOrderConfig(state)
+    },
+    switchAssets: (state: LimitOrderInputState) => {
+      baseReducers.switchAssets(state)
+      resetLimitOrderConfig(state)
+    },
+  }),
```
Comment (gomesalexandre) [inline src/state/slices/limitOrderInputSlice/limitOrderInputSlice.ts:87]: 💜

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
E2E tested locally, LGTM minus akschual price still resetting

- Limit slice state does not leak into swapper and swapper is happy after using limit functionality ✅ 

https://jam.dev/c/e774c8ea-ce6f-4041-8076-b184608fe169

- Custom sell amount is not lost ✅ 
- Pair selection is not lost ✅ 
- Selected price option (market/percentage is not lost) ✅ 
- Actual price resets to default 🚫 i.e regardless of whether market/percentage/custom is used as a price, the actual price always reverts to default

https://jam.dev/c/11ec50fa-4b80-4168-ac29-0b3715d7bbff

**Review by gomesalexandre — APPROVED** [review]
Retested locally, LGTM 🎈 

https://jam.dev/c/a16f02ef-053d-4a69-bf9f-f5f451dbd64d

**Pull request merged**